internal/prometheus/prometheus.go

```diff
@@ -0,0 +1,76 @@
+package prometheus
+
+import (
+	"net/http"
+
+	"github.com/AdguardTeam/AdGuardHome/internal/stats"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Config - Prometheus configuration
+type Config struct {
+	Enabled   bool   `yaml:"enabled"`
+	Namespace string `yaml:"namespace"`
+
+	AnonymizeClientIP bool `yaml:"-"` // anonymize clients' IP addresses - set from DNS config
+}
+
+type Server struct {
+	conf Config
+
+	dnsRequests *prometheus.CounterVec
+	dnsDuration prometheus.Histogram
+}
+
+func Create(config Config) *Server {
+	s := Server{}
+	s.conf.Enabled = config.Enabled
+	s.conf.Namespace = config.Namespace
+
+	if !s.conf.Enabled {
+		return &s
+	}
+
+	if s.conf.Namespace == "" {
+		s.conf.Namespace = "adguard"
+	}
+
+	// Use http.Handle directly to be unauthenticated.
+	http.Handle("/metrics", promhttp.Handler())
```

Comment:
It'd be better to have it listen on a different port & create server mux here instead of registering the handler globally.

Reply:
Fixed!